tests/mysql/macros.rs

```diff
@@ -585,4 +585,28 @@ async fn test_uuid_is_compatible_mariadb() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[derive(PartialEq, Eq, Debug, sqlx::Type)]
+#[sqlx(transparent)]
+struct MyIntNamed {
+    inner: i64,
+}
+
+struct NamedTransparentRecord {
+    id: MyIntNamed,
+}
+
+#[sqlx_macros::test]
+async fn test_named_transparent_struct() -> anyhow::Result<()> {
+    let mut conn = new::<MySql>().await?;
+    let (mut conn, id) = with_test_row(&mut conn).await?;
+
+    let record = sqlx::query_as!(NamedTransparentRecord, "select id as `id: _` from tweet")
+        .fetch_one(&mut *conn)
+        .await?;
+
+    assert_eq!(record.id, MyIntNamed { inner: id.0 });
+
+    Ok(())
+}
```

Comment:
Strictly speaking, this is testing the derive behavior more than the query macro behavior, so it should be moved to `derives.rs`.

When you do that, please add `#[cfg(feature = "macros")]` to the test like the other tests using query macros there. You might also move the struct definitions _into_ the function so they don't emit "unused code" warnings when the feature is off. 

Either that, or just rewrite the test to not use the query macros since that's not really what we're looking to test here.